src/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -354,63 +354,43 @@ public void Socket_BeginConnectV4IPAddressToV4Host_Throws()
-        {
-            DualModeBeginConnect_IPAddressToHost_Helper(IPAddress.Loopback, IPAddress.Loopback, false);
-        }
+        public Task BeginConnectV4IPAddressToV4Host_Success() => DualModeBeginConnect_IPAddressToHost_Helper(IPAddress.Loopback, IPAddress.Loopback, false);
 
         [Fact]
-        public void BeginConnectV6IPAddressToV6Host_Success()
-        {
-            DualModeBeginConnect_IPAddressToHost_Helper(IPAddress.IPv6Loopback, IPAddress.IPv6Loopback, false);
-        }
+        public Task BeginConnectV6IPAddressToV6Host_Success() => DualModeBeginConnect_IPAddressToHost_Helper(IPAddress.IPv6Loopback, IPAddress.IPv6Loopback, false);
 
         [Fact]
-        public void BeginConnectV4IPAddressToV6Host_Fails()
-        {
-            DualModeBeginConnect_IPAddressToHost_Fails_Helper(IPAddress.Loopback, IPAddress.IPv6Loopback);
-        }
+        public Task BeginConnectV4IPAddressToV6Host_Fails() => DualModeBeginConnect_IPAddressToHost_Fails_Helper(IPAddress.Loopback, IPAddress.IPv6Loopback);
 
         [Fact]
-        public void BeginConnectV6IPAddressToV4Host_Fails()
-        {
-            DualModeBeginConnect_IPAddressToHost_Fails_Helper(IPAddress.IPv6Loopback, IPAddress.Loopback);
-        }
+        public Task BeginConnectV6IPAddressToV4Host_Fails() => DualModeBeginConnect_IPAddressToHost_Fails_Helper(IPAddress.IPv6Loopback, IPAddress.Loopback);
 
         [Fact]
-        public void BeginConnectV4IPAddressToDualHost_Success()
-        {
-            DualModeBeginConnect_IPAddressToHost_Helper(IPAddress.Loopback, IPAddress.IPv6Any, true);
-        }
+        public Task BeginConnectV4IPAddressToDualHost_Success() => DualModeBeginConnect_IPAddressToHost_Helper(IPAddress.Loopback, IPAddress.IPv6Any, true);
 
         [Fact]
-        public void BeginConnectV6IPAddressToDualHost_Success()
-        {
-            DualModeBeginConnect_IPAddressToHost_Helper(IPAddress.IPv6Loopback, IPAddress.IPv6Any, true);
-        }
+        public Task BeginConnectV6IPAddressToDualHost_Success() => DualModeBeginConnect_IPAddressToHost_Helper(IPAddress.IPv6Loopback, IPAddress.IPv6Any, true);
 
-        private void DualModeBeginConnect_IPAddressToHost_Helper(IPAddress connectTo, IPAddress listenOn, bool dualModeServer)
+        private async Task DualModeBeginConnect_IPAddressToHost_Helper(IPAddress connectTo, IPAddress listenOn, bool dualModeServer)
         {
-            int port;
             using (Socket socket = new Socket(SocketType.Stream, ProtocolType.Tcp))
-            using (SocketServer server = new SocketServer(_log, listenOn, dualModeServer, out port))
+            using (SocketServer server = new SocketServer(_log, listenOn, dualModeServer, out int port))
             {
-                IAsyncResult async = socket.BeginConnect(connectTo, port, null, null);
-                socket.EndConnect(async);
+                await Task.Factory.FromAsync(socket.BeginConnect(connectTo, port, null, null), socket.EndConnect);
```

Comment:
FWIW, this overload of FromAsync is much less efficient than if you used the one that took two delegates:
```C#
await Task.Factory.FromAsync(socket.BeginConnect, socket.EndConnect, port, null);
```
as the overload you've used needs to use ThreadPool.RegisterWaitForSingleObject.  Not a big deal for test code.

Reply:
I have changed all of these. Might as well make the tests run as fast as possible seeing as there are like 12m on a outerloop.